packages/edit-post/src/components/editor-regions/style.scss

```diff
@@ -54,6 +54,7 @@
 	bottom: 0;
 	left: 0;
 	background: $white;
+	overflow: auto;
```

Comment:
Ohhh, now I get it now!

While this seems to fix the issue if you test on the browser, there's a weird issue in Safari iOS (specially old versions) where the only scrollable element should be the `html` element otherwise the sticky behavior doesn't work properly.

I think the fix should be to make sure the html height increases with the sidebar allowing it to scroll. This also might make the z-index fix for the sidebar header unecessary.

Anyway this is a complex topic that only @jasmussen understands properly :P so I'd prefer to wait for his feedback. He also has this PR open that I think is related #18686